test/testCppFile.js

```diff
@@ -0,0 +1,840 @@
+            '        ResBundleAdaptor::Instance().getLocString("Switch to the game console") + "\"";\n' +
+            '        i18n_yes    = ResBundleAdaptor::Instance().getLocString("Network settings");\n' +
+            '        i18n_no     = ResBundleAdaptor::Instance().getLocString("Cancel");\n' +
+            '    } else {\n' +
+            '    i18n_msg = ResBundleAdaptor::Instance().getLocString("Network is not connected. Do you want to go to Network Settings now to check the connection?");\n' +
+            '\n' +
+            '    i18n_yes    = ResBundleAdaptor::Instance().getLocString("Yes");\n' +
+            '    i18n_no     = ResBundleAdaptor::Instance().getLocString("No");\n' +
+            '}\n');
+
+        var set = cppf.getTranslationSet();
+        test.ok(set);
+
+        test.equal(set.size(), 11);
+
+        var r = set.getBySource("Yes");
+        test.ok(r);
+        test.equal(r.getSource(), "Yes");
+        test.equal(r.getKey(), "Yes");
+
+        r = set.getBySource("Brighten/dim the screen");
+        test.ok(r);
+        test.equal(r.getSource(), "Brighten/dim the screen");
+        test.equal(r.getKey(), "Brighten/dim the screen");
+
+        r = set.getBySource("Channel 20 (channel number)");
+        test.ok(r);
+        test.equal(r.getSource(), "Channel 20 (channel number)");
+        test.equal(r.getKey(), "Channel 20 (channel number)");
+
+        test.done();
+    },
+
+    testCppFileExtractFile: function(test) {
+        test.expect(11);
+
+        var cppf = new CppFile({
+            project: p,
+            pathName: "./t1.cpp",
+            type: cppft
+        });
+        test.ok(cppf);
+
+        // should read the file
+        cppf.extract();
+        var set = cppf.getTranslationSet();
+        test.equal(set.size(), 12);
+
+        var r = set.getBySource("Network settings");
+        test.ok(r);
+        test.equal(r.getSource(), "Network settings");
+        test.equal(r.getKey(), "Network settings");
+
+        var r = set.getBy({
+            reskey: "Search 'OOO(search keyword)' on YouTube"
+        });
+        test.ok(r);
+        test.equal(r[0].getSource(), "Search 'OOO(search keyword)' on YouTube");
+        test.equal(r[0].getKey(), "Search 'OOO(search keyword)' on YouTube");
+
+        var r = set.getBy({
+            reskey: "Switch to \"the game\" console"
+        });
+        test.ok(r);
+        test.equal(r[0].getSource(), "Switch to \"the game\" console");
+        test.equal(r[0].getKey(), "Switch to \"the game\" console");
+
+        test.done();
+    },
+
+    testCppFileExtractFile2: function(test) {
+        test.expect(11);
+
+        var cppf = new CppFile({
+            project: p,
+            pathName: "./t2.cpp",
+            type: cppft
+        });
+        test.ok(cppf);
+
+        // should read the file
+        cppf.extract();
+        var set = cppf.getTranslationSet();
+        test.equal(set.size(), 3);
+
+        var r = set.getBySource("Your\n Welcome");
```

Comment:
Minor nitpick: it should be "You're Welcome". You're = You are. Your = possessive.

Reply:
Oh, you're right. Thanks for the correction :)